From f078fda6ea661e7ebcfbef82632f1da7c45633dd Mon Sep 17 00:00:00 2001
From: kevin shen <b00984@freescale.com>
Date: Fri, 13 Jun 2008 10:51:38 +0800
Subject: [PATCH] ENGR00079975 add FM support for mx35

add FM platform data for mx35

also change mdelay to msleep in si4702_reset function
of mx31 platform

Signed-off-by: Kevin Shen <b00984@freescale.com>
---
 arch/arm/mach-mx3/mx3_3stack.c   |    4 ++--
 arch/arm/mach-mx35/mx35_3stack.c |   37 +++++++++++++++++++++++++++++++++++++
 2 files changed, 39 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx3/mx3_3stack.c b/arch/arm/mach-mx3/mx3_3stack.c
index b534692..3a0a1ca 100644
--- a/arch/arm/mach-mx3/mx3_3stack.c
+++ b/arch/arm/mach-mx3/mx3_3stack.c
@@ -235,9 +235,9 @@ struct mxc_tvout_platform_data tvout_data = {
 void si4702_reset(void)
 {
 	mxc_set_gpio_dataout(MX31_PIN_SRST0, 0);
-	mdelay(100);
+	msleep(100);
 	mxc_set_gpio_dataout(MX31_PIN_SRST0, 1);
-	mdelay(100);
+	msleep(100);
 }
 
 void si4702_clock_ctl(int flag)
diff --git a/arch/arm/mach-mx35/mx35_3stack.c b/arch/arm/mach-mx35/mx35_3stack.c
index 6662e58..ef5ef35 100644
--- a/arch/arm/mach-mx35/mx35_3stack.c
+++ b/arch/arm/mach-mx35/mx35_3stack.c
@@ -25,6 +25,7 @@
 #include <linux/mtd/mtd.h>
 #include <linux/mtd/map.h>
 #include <linux/mtd/partitions.h>
+#include <linux/delay.h>
 
 #include <asm/mach/flash.h>
 #endif
@@ -38,6 +39,7 @@
 #include <asm/arch/memory.h>
 #include <asm/arch/gpio.h>
 #include <asm/arch/mmc.h>
+#include <asm/arch/pmic_external.h>
 
 #include "board-mx35_3stack.h"
 #include "crm_regs.h"
@@ -268,6 +270,36 @@ static struct mxc_camera_platform_data camera_data = {
 	.mclk = 24000000,
 };
 
+void si4702_reset(void)
+{
+	pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_1, 4, 0);
+	msleep(100);
+	pmic_gpio_set_bit_val(MCU_GPIO_REG_RESET_1, 4, 1);
+	msleep(100);
+}
+
+void si4702_clock_ctl(int flag)
+{
+	pmic_gpio_set_bit_val(MCU_GPIO_REG_GPIO_CONTROL_1, 7, flag);
+}
+
+static void si4702_gpio_get(void)
+{
+}
+
+static void si4702_gpio_put(void)
+{
+}
+
+static struct mxc_fm_platform_data si4702_data = {
+	.reg_vio = "SW1",
+	.reg_vdd = "SW2",
+	.gpio_get = si4702_gpio_get,
+	.gpio_put = si4702_gpio_put,
+	.reset = si4702_reset,
+	.clock_ctl = si4702_clock_ctl,
+};
+
 static struct i2c_board_info mxc_i2c_board_info[] __initdata = {
 	{
 	 .driver_name = "mc9sdz60",
@@ -298,6 +330,11 @@ static struct i2c_board_info mxc_i2c_board_info[] __initdata = {
 	 .addr = 0x55,
 	 },
 #endif
+	{
+	 .driver_name = "si4702",
+	 .addr = 0x10,
+	 .platform_data = (void *)&si4702_data,
+	 },
 };
 
 #if  defined(CONFIG_SMSC911X) || defined(CONFIG_SMSC911X_MODULE)
-- 
1.5.4.4

